src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -323,7 +323,12 @@ public void run()
             }
           }
           else {
-            register(channel, SelectionKey.OP_CONNECT, l);
+            /*
```

Comment:
After 3 hours running tests OP_READ was fired without OP_CONNECT. There are two possible solutions, we may either try to register OP_CONNECT in the interestOps and iterate over keys looking for keys with OP_CONNECT and call finishConnect() for all such keys or register OP_CONNECT and OP_READ and call ClientListener.connected() if OP_READ is fired without OP_CONNECT.